core/domain/collection_services.py

```diff
@@ -1132,17 +1115,11 @@ def load_demo(collection_id):
     """
     delete_demo(collection_id)
 
-    if not collection_domain.Collection.is_demo_collection_id(collection_id):
```

Comment:
This is removed as this condition is checked in delete_demo() which is called just above this line